### DISCUSSION TITLE
Can I use postgres LISTEN NOTIFY (using pg_notify) in EdgeDB?

### DISCUSSION THREAD

**Discussion by clmnin** [geldata/gel#1984] [Q&A] [answered]
I'm building a real time app using EdgeDB and wanted to get notified when changes occur in certain tables / types

Postgres has build in support for pub sub and I'm sure these will come to EdgeDB in due time (asyncpg has `add_listener`). But is there a mechanism to achieve this in the current build? Can I connect to the underlying `pg` in a different thread to notify me of changes?

Here is a SO post related to my need - [Link](https://stackoverflow.com/questions/40507599/real-time-database-messaging?noredirect=1&lq=1)

**Comment by 1st1** [comment, marked as answer]
We don't have that supported yet and it will likely happen only after 1.0 release. That said it's on our list.

**Reply by 1st1** [reply]
> Can I connect to the underlining pg in a different thread to notify me of changes?

We don't expose the underlying postgres instance. If run in cloud, this is possible to do, although it will be very hacky. A solution I'd suggest to looking into right now is using a dedicated message broker, like RabbitMQ.
